backend/lib/db.js

```diff
@@ -28,35 +28,15 @@ class DataBase {
     }
 
     getAll(collection, sortParam = {}) {
-        this.db.collection(collection).find({}).sort(sortParam).toArray((err, result) => {
-            if (err) {
-                return err
-            }
-            return result
-        })
+        return this.db.collection(collection).find({}).sort(sortParam).toArray()
     }
 
     getByEmail(collection, email) {
-        this.db.collection(collection).find({ email: email.email }).toArray((err, result) => {
-            if (err) {
-                console.log('getByEmail err', err)
-                return err
-            }
-            if (result.length == 0) {
-                return reject('Not found by email')
-            }
-            return result[0]
-        })
+        return this.db.collection(collection).find({ email: email.email }).toArray()
     }
 
     getById(collection, id) {
-        this.db.collection(collection).find({ _id: new ObjectID(id) }).toArray((err, result) => {
-            if (err) {
-                console.log('getById err', err)
-                return reject(err)
-            }
-            return resolve(result[0])
-        })
+        return this.db.collection(collection).find({ _id: new ObjectID(id) }).toArray()
```

Comment:
You can replace this one with `this.db.collection(collection).findOne` instead of `find`, looking at the original code.

Reply:
+1